### PR TITLE
[BACKLOG-6588] - creating UpdateAttribute annotation

### PR DESCRIPTION
**Pull request by kurtwalker**

### CODE REVIEW

**Comment by lucboudreau** [comment]
http://10.100.11.222:8080/job/wingman-osgi/33/

**Comment by lucboudreau** [comment]
Error building command: /home/devuser/assembly-0.0.1/source/modeler-base-270-c21c502f-8bb7-425f-b4b9-15a0e9e041a9/pom.xml (No such file or directory)

**Comment by lucboudreau** [comment]
Build Commands:
==============
Build:
------
1. ant -Dpentaho.coding.standards.version=1.0.4 -Djunit.jvmargs=-XX:PermSize=256m clean-all resolve jacoco jacoco-integration checkstyle publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  src/org/pentaho/agilebi/modeler/messages/messages_en_US.properties
  src/org/pentaho/agilebi/modeler/models/annotations/ModelAnnotation.java
  src/org/pentaho/agilebi/modeler/models/annotations/UpdateAttribute.java
  src/org/pentaho/agilebi/modeler/models/annotations/util/MondrianSchemaHandler.java
  test-res/updateattribute.mondrian.xml
  test-src/org/pentaho/agilebi/modeler/models/annotations/ModelAnnotationTest.java
  test-src/org/pentaho/agilebi/modeler/models/annotations/UpdateAttributeTest.java










Unit test coverage change:
============
org.pentaho.agilebi.modeler.models.annotations.util.MondrianSchemaHandler:
 * instructionPercentCoverageChange : +0.01%
 * branchPercentCoverageChange : +1.14%
 * linePercentCoverageChange : +0.02%
 * complexityPercentCoverageChange : +1.78%
    
org.pentaho.agilebi.modeler.models.annotations.ModelAnnotation.Type:
 * instructionPercentCoverageChange : +0.74%
 * linePercentCoverageChange : +0.99%
    
org.pentaho.agilebi.modeler.models.annotations.UpdateAttribute:
 * branchPercentCoverage :90.00%
 * linePercentCoverage :89.47%
 * methodPercentCoverage :82.35%
 * complexityPercentCoverage :81.82%
 * instructionPercentCoverage :95.45%
    
Integration test coverage change:
============
org.pentaho.agilebi.modeler.models.annotations.util.MondrianSchemaHandler:
 * instructionPercentCoverageChange :**-1.37%**
 * branchPercentCoverageChange :**-1.51%**
 * linePercentCoverageChange :**-1.88%**
 * complexityPercentCoverageChange :**-2.09%**
 * methodPercentCoverageChange :**-4.90%**
    
org.pentaho.agilebi.modeler.models.annotations.ModelAnnotation.Type:
 * instructionPercentCoverageChange : +1.13%
 * linePercentCoverageChange : +1.19%

**Comment by kurtwalker** [comment]
@rfellows @pminutillo @DFieldFL please review

File: src/org/pentaho/agilebi/modeler/models/annotations/UpdateAttribute.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
+/*!
+ * PENTAHO CORPORATION PROPRIETARY AND CONFIDENTIAL
+ *
+ * Copyright 2002 - 2014 Pentaho Corporation (Pentaho). All rights reserved.
```
Comment (DFieldFL) [inline src/org/pentaho/agilebi/modeler/models/annotations/UpdateAttribute.java]: update copyright to 2016

File: test-src/org/pentaho/agilebi/modeler/models/annotations/UpdateAttributeTest.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
+/*!
+ * PENTAHO CORPORATION PROPRIETARY AND CONFIDENTIAL
+ *
+ * Copyright 2002 - 2014 Pentaho Corporation (Pentaho). All rights reserved.
```
Comment (DFieldFL) [inline test-src/org/pentaho/agilebi/modeler/models/annotations/UpdateAttributeTest.java]: copyright

File: src/org/pentaho/agilebi/modeler/models/annotations/UpdateAttribute.java [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+  @ModelProperty( id = HIERARCHY_ID, name = HIERARCHY_NAME, order = HIERARCHY_ORDER )
+  private java.lang.String hierarchy;
+
+  @MetaStoreAttribute
+  @ModelProperty( id = LEVEL_ID, name = LEVEL_NAME, order = LEVEL_ORDER )
+  private String level;
+
+  @Override public boolean apply( final ModelerWorkspace workspace, final IMetaStore metaStore )
+    throws ModelerException {
+    return false;
```
Comment (DFieldFL) [inline src/org/pentaho/agilebi/modeler/models/annotations/UpdateAttribute.java]: reason we don't have an implementation for this?

Comment (kurtwalker) [inline src/org/pentaho/agilebi/modeler/models/annotations/UpdateAttribute.java]: We won't be using this annotation in the data refinery

**Comment by kurtwalker** [comment]
copyrights updated

**Pull request merged**